crux/src/Crux.hs

```diff
@@ -117,53 +128,93 @@ showVersion l = outputLn ("crux version: " ++ Crux.version ++ ", " ++
+    "uninterpreted" -> withOnlineBackendFM FloatUninterpretedRepr
     "default" ->
-      case (solver cruxOpts) of
-        s@"cvc4" -> withSolver FloatRealRepr s
-        s@"yices" -> withSolver FloatRealRepr s
-        s@"z3" -> withSolver FloatIEEERepr s
-        s -> fail $ "unknown solver: " ++ s ++ "; expepected of one [cvc4|yices|z3]."
-    fm -> fail $ "unknown floating-point mode: " ++ fm ++ "; expepected of one [real|ieee|uninterpreted|default]."
+      case selectedSolver of
+        CCS.YicesOnline -> withOnlineBackendFM FloatRealRepr
+        CCS.CVC4Online -> withOnlineBackendFM FloatRealRepr
+        CCS.STPOnline -> withOnlineBackendFM FloatRealRepr
+        CCS.Z3Online -> withOnlineBackendFM FloatIEEERepr
+    fm -> fail ("Unknown floating point mode: " ++ fm ++ "; expected one of [real|ieee|uninterpreted|default]")
   where
-  unsatCores | yicesMCSat cruxOpts = NoUnsatFeatures
-             | otherwise           = ProduceUnsatCores
-  withSolver fpMode "cvc4" =
-    withCVC4OnlineBackend fpMode nonceGen ProduceUnsatCores $ \sym->
-      do case goalTimeout cruxOpts of
-           Just s -> symCfg sym cvc4Timeout (floor (s * 1000))
-           Nothing -> return ()
-         f sym
-  withSolver fpMode "yices" =
-    withYicesOnlineBackend fpMode nonceGen unsatCores $ \sym ->
-      do symCfg sym yicesEnableMCSat (yicesMCSat cruxOpts)
-         case goalTimeout cruxOpts of
-           Just s -> symCfg sym yicesGoalTimeout (floor s)
-           Nothing -> return ()
-         f sym
-  withSolver fpMode "z3" =
-    withZ3OnlineBackend fpMode nonceGen ProduceUnsatCores $ \sym->
-      do case goalTimeout cruxOpts of
-           Just s -> symCfg sym z3Timeout (floor (s * 1000))
-           Nothing -> return ()
-         f sym
-  withSolver _fpMode other =
-    fail $ "unknown solver: " ++ other ++ "; expepected of one [cvc4|yices|z3]."
+    unsatCores | yicesMCSat cruxOpts = NoUnsatFeatures
+               | otherwise           = ProduceUnsatCores
+
+    withOnlineBackendFM floatRepr =
+      case selectedSolver of
+        CCS.YicesOnline -> withYicesOnlineBackend floatRepr nonceGen unsatCores $ \sym -> do
+          symCfg sym yicesEnableMCSat (yicesMCSat cruxOpts)
+          case goalTimeout cruxOpts of
+            Just s -> symCfg sym yicesGoalTimeout (floor s)
+            Nothing -> return ()
+          k floatRepr sym
+        CCS.CVC4Online -> withCVC4OnlineBackend floatRepr nonceGen ProduceUnsatCores $ \sym -> do
+          case goalTimeout cruxOpts of
+            Just s -> symCfg sym cvc4Timeout (floor (s * 1000))
+            Nothing -> return ()
+          k floatRepr sym
+        CCS.Z3Online -> withZ3OnlineBackend floatRepr nonceGen ProduceUnsatCores $ \sym -> do
+          case goalTimeout cruxOpts of
+            Just s -> symCfg sym z3Timeout (floor (s * 1000))
+            Nothing -> return ()
+          k floatRepr sym
+        CCS.STPOnline ->
+          -- We don't have a timeout option for STP
```

Comment:
Issue a warning, perhaps, if `goalTimeout` is configured but we can't accomidate?